src/map/terrain.h

```diff
@@ -20,18 +20,23 @@ enum {
     TERRAIN_FOUNTAIN_RANGE = 0x2000,
     TERRAIN_WALL = 0x4000,
     TERRAIN_GATEHOUSE = 0x8000,
+    TERRAIN_ORIGINALLY_TREE = 0x10000,
+    TERRAIN_HIGHWAY_TOP_LEFT = 0x20000,
```

Comment:
If this terrain division really needed? If so, why?

I could check but I'm using the phone and I don't really have the time to understand the overall code structure.

Reply:
I think this is easier to understand with a screenshot:
![image](https://user-images.githubusercontent.com/2081059/194786443-70ba96a5-95da-463e-b684-01c5084ee321.png)
What we're doing here is for each 2x2 highway tile (the blue box) that gets placed, we set the underlying terrain based on the position of each highway tile within the 2x2 tile. With that, we're able to use the following code to figure out if the direction we want to move in would be on the right side of the highway.
```C
static const int HIGHWAY_DIRECTIONS[] = { 
    TERRAIN_HIGHWAY_TOP_RIGHT | TERRAIN_HIGHWAY_BOTTOM_RIGHT, // up
    TERRAIN_HIGHWAY_BOTTOM_LEFT | TERRAIN_HIGHWAY_BOTTOM_RIGHT, // right
    TERRAIN_HIGHWAY_TOP_LEFT | TERRAIN_HIGHWAY_BOTTOM_LEFT, // down
    TERRAIN_HIGHWAY_TOP_LEFT | TERRAIN_HIGHWAY_TOP_RIGHT, // left
};
```